### PR TITLE
Include schemas for bare metal API

### PR DESCRIPTION
**Pull request by guioliveirabh**
Include schemas and documentation for bare metal API

### Fixes

- [EXDRHUB-1010](https://issues.redhat.com/browse/EXDRHUB-1010)

### Checklist

- [ ] Related tests were updated
- [x] Related documentation was updated
- [x] OpenAPI file was updated
- [x] Run `tox`, no tests failed

### CODE REVIEW

File: bare_metal_how_to.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+$ sudo grep -e password /etc/ironic/ironic.conf
+```
+
+#### Insertion on Resource Hub
+
+```python
+requests.post(
+    BASE_URL + 'handler',
+    json=dict(
+        name='setup1',
```
Comment (cbouchar) [inline bare_metal_how_to.md:34]: did you want setup<N>  ?

File: bare_metal_how_to.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+        arch='x86_64',
+        location_id='<LOCATION_ID>',
+        user_name='admin',
+        password='...',
+        base_url='http://dell-r730-025.dsal.lab.eng.rdu2.redhat.com:6385',
```
Comment (cbouchar) [inline bare_metal_how_to.md:39]: We shouldn't publicize RedHat host names in github.  Make something up like example.yourcompany.com

File: bare_metal_how_to.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+Resource Hub will run some checks to validate the Ironic access.
+
+### Insert host(s) on Resource Hub
+
+```python
+requests.post(
+    BASE_URL + 'host',
+    json=dict(
+        handler_id='<IRONIC_ID>',
+        name='dell-r630-003.dsal.lab.eng.rdu2.redhat.com',
```
Comment (cbouchar) [inline bare_metal_how_to.md:54]: Again no hostnames publicized

File: bare_metal_how_to.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+### Insert image(s) on Resource Hub
+
+```python
+requests.post(
+    BASE_URL + 'image',
+    json=dict(
+        base_os='Red_Hat',
+        version='9.0',
+        arch='x86_64',
+        download_url='http://download.hosts.prod.upshift.rdu2.redhat.com/rhel-9/rel-eng/RHEL-9/RHEL-9.0.0-20220420.0/compose/BaseOS/x86_64/iso/RHEL-9.0.0-20220420.0-x86_64-dvd1.iso',
```
Comment (cbouchar) [inline bare_metal_how_to.md:84]: Shouldn't publicize RHEL images locations outside Redhat.  Perhaps use Centos/Fedora instead.

File: bare_metal_how_to.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+timezone UTC --utc
+#platform x86_64
+reboot
+
+## with text, anaconda asks for user input
+cmdline
+#text
+
+url --url {{ resource_hub.image_url }}
+network --bootproto=dhcp --hostname="{{ hostname }}" --device 20:04:0F:E9:04:78 --activate
```
Comment (cbouchar) [inline bare_metal_how_to.md:128]: Don't publicize mac addresses

Comment (freedgit) [reply bare_metal_how_to.md:128]: Dude you got a Dell

File: src/rhub/openapi/bare_metal/common.yml [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+      - maintenance
+      - unavailable
+
+  ImageBaseOS:
+    type: string
+    description: Image's base OS
+    enum:
+      - CentOS
+      - Fedora
+      - Red_Hat
```
Comment (cbouchar) [inline src/rhub/openapi/bare_metal/common.yml:33]: Should this be RHEL?

**Review by freedgit — COMMENTED** [review]
I second the "anonymization" requests by Carol. I wonder if we could have this running in a container instead of having Centos Stream? (by using host networking, that is.)

File: src/rhub/openapi/common.yml [hunk 7]
```diff
@@ -1,9 +1,57 @@
...
+    type: boolean
+    default: true
+
   ID:
     type: integer
     minimum: 1
 
+  MACAddress:
+    description: MAC address
+    example: 11:BB:CC:00:87:EF
```
Comment (cbouchar) [inline src/rhub/openapi/common.yml:13]: Is this a real mac address?  If so, replace is 11:22:33:44:55:66

**Review by freedgit — DISMISSED** [review]
LGTM now

File: bare_metal_how_to.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+
+<!-- TODO: update after https://gitlab.cee.redhat.com/pnt-resource-hub/rhub-baremetal/-/merge_requests/11 -->
+
+Run playbook available on: https://gitlab.cee.redhat.com/pnt-resource-hub/rhub-baremetal
+
+```shell
+$ ansible-playbook -i inventory/dsal.yml playbooks/bifrost_setup.yml --extra-vars "target=setup<N>"
+```
+
+Get the password:
```
Comment (cbouchar) [inline bare_metal_how_to.md:22]: Might want to explain why they need to get this password.  Where is it needed later?

File: src/rhub/api/bare_metal/host.py [hunk 9]
```diff
@@ -38,6 +28,18 @@ def host_create(body):
     return host.to_dict_with_super()
 
 
+def host_create_ipmi(body):
```
Comment (cbouchar) [inline src/rhub/api/bare_metal/host.py:31]: There are other hardware drivers that I listed in the HW wish list.  Of course we need equipment to add that code.

File: src/rhub/openapi/bare_metal/image.yml [hunk 10]
```diff
@@ -0,0 +1,182 @@
...
+      arch:
+        allOf:
+          - $ref: 'common.yml#/components/BareMetalArch'
+      legacy_bios:
+        allOf:
+          - $ref: '../common.yml#/model/BooleanDefaultTrue'
+      uefi:
+        allOf:
+          - $ref: '../common.yml#/model/BooleanDefaultTrue'
+      secure_boot:
```
Comment (cbouchar) [inline src/rhub/openapi/bare_metal/image.yml:25]: Do you think we should be consistent and previx with uefi_secure_boot?  Up to you.

**Review by cbouchar — APPROVED** [review]

**Pull request merged**